SignalRServiceBenchmarkPlugin/utils/Commander/Program.cs

```diff
@@ -22,242 +22,8 @@ public static void Main(string[] args)
-            ScpClient masterScpClient = null;
-            List<ScpClient> slaveScpClients = null;
-
-            try
-            {
-                // Create scp client
-                appServerScpClient = new ScpClient(appserverHostname, username, password);
-                masterScpClient = new ScpClient(masterHostname, username, password);
-                slaveScpClients = (from slave in slaveHostnames
-                                   let hostname = slave.Split(':')[0]
-                                   select new ScpClient(hostname, username, password)).ToList();
-
-                // Connect to host
-                var connectTasks = new List<Task>();
-                connectTasks.Add(Task.Run(() => appServerScpClient.Connect()));
-                connectTasks.Add(Task.Run(() => masterScpClient.Connect()));
-                connectTasks.AddRange(from slaveScpClient in slaveScpClients
-                                      select Task.Run(() => slaveScpClient.Connect()));
-                Task.WhenAll(connectTasks).Wait();
-
-                // Copy executables
-                var copyTasks = new List<Task>();
-                copyTasks.Add(Task.Run(() => appServerScpClient.Upload(appserverExecutable, appserverTargetPath)));
-                copyTasks.Add(Task.Run(() => masterScpClient.Upload(masterExecutable, masterTargetPath)));
-                copyTasks.Add(Task.Run(() => masterScpClient.Upload(benchmarkConfiguration, benchmarkConfigurationTargetPath)));
-                copyTasks.AddRange(from slaveScpClient in slaveScpClients
-                                   select Task.Run(() => slaveScpClient.Upload(slaveExecutable, slaveTargetPath)));
-                Task.WhenAll(copyTasks).Wait();
-            }
-            finally
-            {
-                // Disconnect
-                appServerScpClient.Disconnect();
-                masterScpClient.Disconnect();
-                slaveScpClients.ForEach(client => client.Disconnect());
-
-                // Dispose
-                appServerScpClient.Dispose();
-                masterScpClient.Dispose();
-                slaveScpClients.ForEach(client => client.Dispose());
-            }
-
-        }
-
-        private static FileInfo PubishExcutable(string projectPath, string baseName)
-        {
-            var publish = Command.Run("dotnet", $"publish -o {baseName}".Split(' '), o => o.WorkingDirectory(projectPath));
-            publish.Wait();
-            if (!publish.Result.Success) throw new Exception(publish.Result.StandardOutput);
-
-            var zip = Command.Run("zip", new string[] { "-r", $"{baseName}.zip", $"{baseName}/" }, o => o.WorkingDirectory(projectPath));
-            zip.Wait();
-            if (!zip.Result.Success) throw new Exception(zip.Result.StandardOutput);
-
-            return new FileInfo(Path.Combine(projectPath, $"{baseName}.zip"));
+            var automationTool = new AutomationTool(argsOption);
```

Comment:
Refactor the code, please take a look @ccic

Reply:
Yes, it looks clean.